ports/stm/common-hal/busio/I2C.c

```diff
@@ -59,33 +59,39 @@ void i2c_reset(void) {
 void common_hal_busio_i2c_construct(busio_i2c_obj_t *self,
         const mcu_pin_obj_t* scl, const mcu_pin_obj_t* sda, uint32_t frequency, uint32_t timeout) {
 
-    //match pins to I2C objects
+    // Match pins to I2C objects
     I2C_TypeDef * I2Cx;
     uint8_t sda_len = MP_ARRAY_SIZE(mcu_i2c_sda_list);
     uint8_t scl_len = MP_ARRAY_SIZE(mcu_i2c_scl_list);
     bool i2c_taken = false;
+    bool search_done = false;
 
     for (uint i = 0; i < sda_len; i++) {
         if (mcu_i2c_sda_list[i].pin == sda) {
             for (uint j = 0; j < scl_len; j++) {
                 if ((mcu_i2c_scl_list[j].pin == scl)
-                    && (mcu_i2c_scl_list[j].i2c_index == mcu_i2c_sda_list[i].i2c_index)) {
-                    //keep looking if the I2C is taken, could be another SCL that works
-                    if (reserved_i2c[mcu_i2c_scl_list[i].i2c_index - 1]) {
+                    && (mcu_i2c_scl_list[j].periph_index == mcu_i2c_sda_list[i].periph_index)) {
+                    // Keep looking if the I2C is taken, could be another SCL that works
+                    if (reserved_i2c[mcu_i2c_scl_list[j].periph_index - 1]) {
                         i2c_taken = true;
                         continue;
                     }
                     self->scl = &mcu_i2c_scl_list[j];
                     self->sda = &mcu_i2c_sda_list[i];
+                    // Multi-level break here, or it'll pick the highest numbered peripheral (inefficient)
+                    search_done = true;
                     break;
                 }
             }
+            if (search_done) {
```

Comment:
```suggestion
            if (self->scl != NULL) {
```

No need for an extra bool